File: assets/demos/params.pl
sub foo { 
  my ($x,$y) = @_; 
  print "foo.x:$x\n"; 
  print "foo.y:$y\n"; 
}

$x = 1;
print "++x, ++x\n";
foo(++$x, ++$x);
print "x:$x\n\n";

$x = 1;
print "x++, x++\n";
foo($x++, $x++);
print "x:$x\n\n";

$x = 1;
print "x++, ++x\n";
foo($x++, ++$x);
print "x:$x\n\n";

$x = 1;
print "++x, x++\n";
foo(++$x, $x++);
print "x:$x\n\n";

File: assets/demos/params.c
#include <stdio.h>

void foo(int x, int y) {
  printf("foo.x:%d\nfoo.y:%d\n", x, y); 
}

int main() {
  int x;

  x = 1;
  printf("++x ++x\n");
  foo(++x, ++x);
  printf("x:%d\n\n", x);

  x = 1;
  printf("x++ x++\n");
  foo(x++, x++);
  printf("x:%d\n\n", x);

  x = 1;
  printf("x++ ++x\n");
  foo(x++, ++x);
  printf("x:%d\n\n", x);

  x = 1;
  printf("++x x++\n");
  foo(++x, x++);
  printf("x:%d\n\n", x);
}
